WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.UniversalWindows/HttpRest/RestDelete.cs

```diff
@@ -56,9 +57,19 @@ public partial class DevicePortal
                 {
                     if (!response.IsSuccessStatusCode)
                     {
+                        // If this isn't a retry and it failed due to a bad CSRF
+                        // token, issue a GET to refresh the token and then retry.
+                        if (allowRetry && response.StatusCode == HttpStatusCode.Forbidden && response.ReasonPhrase.Equals("CSRF Token Invalid"))
+                        {
+                            await this.GetOperatingSystemInformation();
```

Comment:
I'd love to see something like:
await RefreshCsrfToken();

private async Task RefreshCsrfToken()
{
    await this.GetMachineName();
}

The idea is to as small of a request as we can and make it easy to update all locations at once if we find a better way to handle this.